interfaces/policy/basedeclaration_test.go

```diff
@@ -1082,6 +1082,27 @@ func (s *baseDeclSuite) TestPlugInstallation(c *C) {
 			}
 		}
 	}
+
+	// test desktop specially
+	ic := s.installPlugCand(c, "desktop", snap.TypeApp, `name: desktop
+version: 0
+type: app
+plugs:
+  desktop:
+`)
+	err := ic.Check()
+	c.Assert(err, IsNil)
+	// desktop-file-ids can only be set by the store
+	ic = s.installPlugCand(c, "desktop", snap.TypeApp, `name: desktop
```

Comment:
nitpick: if this doesn't fit in the test's pattern, I'd prefer to put it in its own test to avoid having each test do too much and become hard to understand

Reply:
Thanks for pointing this out, It made me reconsider and created a separate test that actually tests the store-side snap-declaration override.